src/player.cpp

```diff
@@ -1810,6 +1810,14 @@ bool player::digging() const
     return false;
 }
 
+bool player::can_reload()
+{
+    if (!weapon.is_gun()) {
+        return false;
+    }
+    return (weapon.charges < weapon.type->gun->clip && get_ammo(weapon.ammo_type()).size() > 0);
```

Comment:
FYI (it's a pre-existing function and therefor not your problem): the logic here is very simple and will miss quite a few cases:

- tools can be reloaded (not sure if this is important here, but for example the tazer is a tool).
- spare magazines, double magazines are not considered.
- reloading an auxiliary gun mod is not considered.
- flags like "NO_RELOAD" and "RELOAD_AND_SHOOT" are not considered.

I'm not even sure whether those things should be considered here.

Btw. it could also be merged with the content of `player::rate_action_reload`.